lib/assert.js

```diff
@@ -885,6 +886,64 @@ assert.ifError = function ifError(err) {
   }
 };
 
+assert.match = function match(string, regexp, message) {
+  if (!isRegExp(regexp)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'regexp', 'RegExp', regexp
+    );
+  }
+  if (typeof string !== 'string' || !RegExpPrototypeTest(regexp, string)) {
+    if (message instanceof Error) {
+      throw message;
+    }
+
+    const generatedMessage = !message;
+    message = message || (typeof string !== 'string' ?
+      'The "string" argument must be of type string. Received type ' +
+        `${typeof string} (${inspect(string)})` :
+      'The input did not match the regular expression ' +
+        `${inspect(regexp)}. Input:\n\n${inspect(string)}\n`);
+    const err = new AssertionError({
+      actual: string,
+      expected: regexp,
+      message,
+      operator: 'match',
+      stackStartFn: match
+    });
+    err.generatedMessage = generatedMessage;
+    throw err;
+  }
+};
+
+assert.notMatch = function notMatch(string, regexp, message) {
```

Comment:
We could align the name with `.doesNotThrow()` as in `doesNotMatch()`

Reply:
> We could align the name with `.doesNotThrow()` as in `doesNotMatch()`

I like that.